src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -856,6 +856,9 @@ NESTED_ENTRY JIT_MonTryEnter_InlineGetThread, _TEXT
         mov                 [rsp + 8h], rcx
         .savereg            rdx, 10h
         mov                 [rsp + 10h], rdx
+	
+        push_nonvol_reg     r12 ; rcx now at [rsp + 10h]
+		
```

Comment:
I do not see a problem with adding extra ifdef in a few more places.

The shortest codepath through this helper is the non-contended case that acquires lock successfully. We should not be making this path longer. Once there is a contention - that will always burn a lot of cycles underneath, accessing stack instead of register is not a problem.